asreview/models/feature_extractors.py

```diff
@@ -46,7 +46,58 @@ def transform(self, X):
         return X[self.columns].fillna("").apply(lambda x: self.sep.join(x), axis=1)
 
 
-class Tfidf(Pipeline):
+class VectorizerPipeline(Pipeline):
+    """Base class for TF-IDF and OneHot feature extraction.
+
+    This class provides a common structure for TF-IDF and OneHot extraction
+    with support for text merging and vectorization.
+    """
+
+    name = None
+    label = None
+
+    def __init__(self, vectorizer_class, **kwargs):
+        text_merger_params = {"columns": ["title", "abstract"]}
+        vectorizer_params = {}
+
+        for key, value in kwargs.items():
+            if key.startswith("text_merger__"):
+                text_merger_params[key.split("__", 1)[1]] = value
+            elif key.startswith("vectorizer__"):
+                vectorizer_params[key.split("__", 1)[1]] = value
+            else:
+                vectorizer_params[key] = value
+
+        if "ngram_range" in vectorizer_params:
+            vectorizer_params["ngram_range"] = tuple(vectorizer_params["ngram_range"])
+
+        super().__init__(
+            [
+                ("text_merger", TextMerger(**text_merger_params)),
+                ("vectorizer", vectorizer_class(**vectorizer_params)),
+            ]
+        )
+
+    def get_params(self, deep=True, instances=False):
+        """Get parameters for this pipeline.
+
+        Parameters
+        ----------
+        deep: bool, default=True
+            If True, will return the parameters for this pipeline and
+            contained subobjects that are estimators.
+        instances: bool, default=False
+            If True, will return the instances of the estimators in the pipeline.
+        """
+        params = super().get_params(deep=deep)
+        if not instances:
+            params.pop("text_merger", None)
+            params.pop("vectorizer", None)
+            params.pop("vectorizer__dtype", None)
+        return params
```

Comment:
I am quite unhappy about these, so I am open to suggestions.
The reason for these pops are:
1. `text_merger` and `vectorizer` are instances, so they are not JSON serializable in the .to_file() function
2. `vectorizer__dtype` is by default a `numpy.float64` which is also not JSON serializable in the .to_file() function